dali/base/dautils.cpp

```diff
@@ -3838,13 +3838,16 @@ class FileReadPropertiesUpdater : public CSimpleInterfaceOf<IFileReadPropertiesU
     // - if curReadCost is 0, it will be calculated using calcFileAccessCost
     virtual cost_type addCostAndNumReads(IDistributedFile * file, stat_type numDiskReads, cost_type curReadCost) override
     {
+        const char * logicalFileName = file->queryLogicalName();
+        if (strsame(logicalFileName, "")) // Skip files without logical names (typically spill files)
```

Comment:
trivial: isEmptyString() is more efficient, and clearer.  (also works on nullptrs)